src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -3184,15 +3184,19 @@ internal virtual ImmutableArray<LabelSymbol> Labels
         /// <param name="patternType">Type to search.</param>
         /// <param name="methodName">Method to search for.</param>
         /// <param name="lookupResult">Passed in for reusability.</param>
+        /// <param name="searchExtensionMethods">True if the lookup should consider extension methods as viable candidates</param>
         /// <param name="warningsOnly">True if failures should result in warnings; false if they should result in errors.</param>
         /// <param name="diagnostics">Populated with binding diagnostics.</param>
         /// <returns>The desired method or null.</returns>
         internal MethodSymbol FindPatternMethod(TypeSymbol patternType, string methodName, LookupResult lookupResult,
-                                                SyntaxNode syntaxExpr, bool warningsOnly, DiagnosticBag diagnostics,
+                                                SyntaxNode syntaxExpr, bool searchExtensionMethods, bool warningsOnly, DiagnosticBag diagnostics,
                                                 SyntaxTree syntaxTree, MessageID messageID)
         {
             Debug.Assert(lookupResult.IsClear);
-            
+
+            //prototype:
+            LookupOptions lookupOptions = (searchExtensionMethods) ? LookupOptions.MustBeInvocableIfMember : LookupOptions.Default;
+
             // Not using LookupOptions.MustBeInvocableMember because we don't want the corresponding lookup error.
```

Comment:
This comment seems relevant. It sounds like there may be an unwanted diagnostic added.